guacamole-common-js/src/main/webapp/modules/Mouse.js

```diff
@@ -273,38 +279,46 @@ Guacamole.Mouse = function(element) {
         // Update overall delta
         scroll_delta += delta;
 
-        // Up
-        while (scroll_delta <= -guac_mouse.scrollThreshold) {
-
-            if (guac_mouse.onmousedown) {
-                guac_mouse.currentState.up = true;
-                guac_mouse.onmousedown(guac_mouse.currentState);
-            }
-
-            if (guac_mouse.onmouseup) {
-                guac_mouse.currentState.up = false;
-                guac_mouse.onmouseup(guac_mouse.currentState);
-            }
-
-            scroll_delta += guac_mouse.scrollThreshold;
-
-        }
-
-        // Down
-        while (scroll_delta >= guac_mouse.scrollThreshold) {
-
-            if (guac_mouse.onmousedown) {
-                guac_mouse.currentState.down = true;
-                guac_mouse.onmousedown(guac_mouse.currentState);
-            }
-
-            if (guac_mouse.onmouseup) {
-                guac_mouse.currentState.down = false;
-                guac_mouse.onmouseup(guac_mouse.currentState);
-            }
-
-            scroll_delta -= guac_mouse.scrollThreshold;
-
+        if (mouse_scroll_timeout == null) {
+            // send actual mouse scroll event after a threshold
+            mouse_scroll_timeout = window.setTimeout(function() {
+
+                // Up
+                if (scroll_delta < 0) {
+                    do {
+                        if (guac_mouse.onmousedown) {
+                            guac_mouse.currentState.up = true;
+                            guac_mouse.onmousedown(guac_mouse.currentState);
+                        }
+
+                        if (guac_mouse.onmouseup) {
+                            guac_mouse.currentState.up = false;
+                            guac_mouse.onmouseup(guac_mouse.currentState);
+                        }
+
+                        scroll_delta += guac_mouse.scrollThreshold;
+                                } while (scroll_delta < 0);
```

Comment:
Please follow the style of the surrounding code. It looks like this is partially indented with tabs, and partially indented with spaces, leading to brain-melting misalignment.

We use a four-space indent across the board - no tabs please.